src/components/Register/Register.js

```diff
@@ -1,26 +1,44 @@
-import React from "react";
-import { EmailInput } from "../EmailInput/EmailInput";
-import { PasswordInput } from "../PasswordInput/PasswordInput";
-import { UsernameInput } from "../UsernameInput/UsernameInput";
+import React, { useContext, useEffect } from "react";
+import { FormInput } from "../FormInput/FormInput";
 import { SaveFormButton } from "../SaveFormButton/SaveFormButton";
 import { FormValidator } from "../../utils/FormValidator/FormValidator";
 import { formSettings } from "../../utils/helpers";
+import * as auth from "../../utils/auth";
+import { FormContext } from "../../contexts/FormContext";
 
 export function Register({
-  handleInputEmail,
-  handleInputPassword,
-  userName,
-  email,
-  password,
-  handleSubmitRegister,
-  handleInputUsername,
-  handleSwitchRegisterToLoginPopup,
+  handleSetRegistration,
   handleSubmitInfoToolTip,
+  handleSwitchPopup,
 }) {
+  const {
+    username: [username, setUsername],
+    email: [email, setEmail],
+    password: [password, setPassword],
+  } = useContext(FormContext);
   const [form, setForm] = React.useState({});
   const formRef = React.useRef();
 
-  React.useEffect(() => {
+  function handleSubmitRegister(e) {
+    e.preventDefault();
+    auth
+      .register({
+        username,
+        email,
+        password,
+      })
+      .then((result) => {
+        if (result && result._id) {
+          handleSetRegistration();
+          handleSubmitInfoToolTip();
+        }
+      })
+      .catch((err) => {
+        console.log(`Error..... ${err}`);
+      });
+  }
+
+  useEffect(() => {
     const validatedForm = new FormValidator(formSettings, formRef.current);
```

Comment:
**Might be better**

that's a `js`-variant of validation. 

If it’s interesting for you here is a custom `validation hook`.

You should put it in a separate file  `useFormAndValidation.js` in folder `hooks` and then import it to the component that needs validation:

```js

import {useState, useCallback} from 'react';

export function useFormAndValidation() {
  const [ values, setValues ] = useState({});
  const [ errors, setErrors ] = useState({});
  const [ isValid, setIsValid ] = useState(true);

  const handleChange = (e) => {
    const {name, value} = e.target
    setValues({...values, [name]: value });
    setErrors({...errors, [name]: e.target.validationMessage});
    setIsValid(e.target.closest('form').checkValidity());
  };



  const resetForm = useCallback((newValues = {}, newErrors = {}, newIsValid = false) => {
    setValues(newValues);
    setErrors(newErrors);
    setIsValid(newIsValid);
  }, [setValues, setErrors, setIsValid]);

  return { values, handleChange, errors, isValid, resetForm, setValues, setIsValid };
}


```

And only one line of code enables validation:

```js

  const {values, handleChange, errors, isValid, setValues, resetForm} = useFormAndValidation()

```

So, you don’t need to make functions for handling inputs, storing values, resetting the form. Everything is already done.